CMakeLists.txt

```diff
@@ -4,8 +4,8 @@ cmake_minimum_required (VERSION 3.8)
 
 # Define the application name and version.
 project (fastq_to_fasta VERSION 1.0.0)
-set (CMAKE_CXX_STANDARD 17)
-set (CMAKE_CXX_STANDARD_REQUIRED ON)
+# set (CMAKE_CXX_STANDARD 17)
+# set (CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
This seems wrong? We actually require cpp20, or cpp17 with concepts.

I have a problem on gcc10 cpp20 linux with it.
It appends and hence overwrite the `-std=c++20` flag I pass.
The resulting options are `-std=c++20 -std=c++1z`, hence it builds in cpp17 mode.

gcc10 on macos graciously adds `-fconcepts`, so it still builds.
On linux, the `-fconcepts` is not added, so it fails.